test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java

```diff
@@ -0,0 +1,580 @@
+						cryptoCreate(TOKEN_TREASURY),
+						cryptoCreate(ANOTHER_USER),
+						tokenCreate(NFTdeleted)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.initialSupply(0)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.adminKey(ADMIN_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						tokenAssociate(ANOTHER_USER, NFTdeleted),
+						mintToken(NFTdeleted, List.of(metadata(FIRST_MEMO), metadata(SECOND_MEMO))),
+						cryptoTransfer(
+								TokenMovement.movingUnique(1L, NFTdeleted)
+										.between(TOKEN_TREASURY, ANOTHER_USER)),
+						tokenDelete(NFTdeleted)
+				).then(
+						cryptoTransfer(
+								TokenMovement.movingUnique(2L, NFTdeleted)
+										.between(TOKEN_TREASURY, ANOTHER_USER))
+								.hasKnownStatus(TOKEN_WAS_DELETED)
+				);
+	}
+
+
+	private HapiApiSpec cannotUnfreezeNftWhenDeleted() {
+		return defaultHapiSpec("cannotUnfreezeNftWhenDeleted")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						newKeyNamed(FREEZE_KEY),
+						newKeyNamed(ADMIN_KEY),
+						cryptoCreate(TOKEN_TREASURY).balance(0L).key(ADMIN_KEY),
+						tokenCreate(NFTdeleted)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.initialSupply(0L)
+								.freezeKey(FREEZE_KEY)
+								.freezeDefault(true)
+								.adminKey(ADMIN_KEY)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+
+				).when(
+						tokenDelete(NFTdeleted)
+				).then(
+						tokenUnfreeze(NFTdeleted, TOKEN_TREASURY)
+								.hasKnownStatus(TOKEN_WAS_DELETED)
+				);
+	}
+
+	private HapiApiSpec cannotFreezeNftWhenDeleted() {
+		return defaultHapiSpec("cannotFreezeNftWhenDeleted")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						newKeyNamed(FREEZE_KEY),
+						newKeyNamed(ADMIN_KEY),
+						cryptoCreate(TOKEN_TREASURY).balance(0L).key(ADMIN_KEY),
+						tokenCreate(NFTdeleted)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.freezeKey(FREEZE_KEY)
+								.adminKey(ADMIN_KEY)
+								.supplyKey(SUPPLY_KEY)
+								.initialSupply(0L)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						tokenDelete(NFTdeleted)
+				).then(
+						tokenFreeze(NFTdeleted, TOKEN_TREASURY)
+								.hasPrecheck(OK)
+								.hasKnownStatus(TOKEN_WAS_DELETED)
+				);
+	}
+
+	private HapiApiSpec cannotDissociateNftWhenDeleted() {
+		return defaultHapiSpec("cannotDissociateNftWhenDeleted")
+				.given(
+						newKeyNamed(ADMIN_KEY),
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY).key(ADMIN_KEY),
+						cryptoCreate(ANOTHER_USER),
+						tokenCreate(NFTdeleted)
+								.initialSupply(0)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.adminKey(ADMIN_KEY)
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(ANOTHER_USER, NFTdeleted)
+				)
+				.when(
+						tokenDelete(NFTdeleted)
+				)
+				.then(
+						tokenDissociate(ANOTHER_USER, NFTdeleted)
+								.hasKnownStatus(SUCCESS)
```

Comment:
We should expect `TOKEN_WAS_DELETED` right?

Reply:
Current implementation gives a SUCCESS for this operation. I will look into more and decide whether it's an issue or not.

Reply:
@qnswirlds seems like this is right behavior, do you think we need more research here?